druid-shell/src/platform/x11/window.rs

```diff
@@ -314,64 +292,37 @@ impl Window {
         xcb::destroy_window(self.app.connection(), self.id);
     }
 
-    fn cairo_surface(&self) -> Result<XCBSurface, Error> {
-        match self.cairo_context.try_borrow() {
-            Ok(ctx) => match ctx.get_target().try_into() {
-                Ok(surface) => Ok(surface),
-                Err(err) => Err(Error::Generic(format!(
-                    "Window::cairo_surface ctx.try_into: {}",
-                    err
-                ))),
-            },
-            Err(err) => Err(Error::BorrowError(format!(
-                "Window::cairo_surface cairo_context: {}",
-                err
-            ))),
-        }
+    fn cairo_surface(&self) -> Result<XCBSurface, anyhow::Error> {
+        self.cairo_context
+            .try_borrow()
+            .context("Window::cairo_surface cairo_context")?
+            .get_target()
+            .try_into()
+            .map_err(|_| anyhow!("Window::cairo_surface try_into"))
     }
 
     fn size(&self) -> Result<Size, Error> {
-        match self.state.try_borrow() {
-            Ok(state) => Ok(state.size),
-            Err(err) => Err(Error::BorrowError(format!("Window::size state: {}", err))),
-        }
+        Ok(self.state.try_borrow().context("Window::size")?.size)
```

Comment:
Let's keep the variable names in the context message like in the old error. Yes right now it's the only varible, but this will allow code growth without having to change things.

Reply:
This (and all the other places where I didn't include a variable name) now uses the macro.